strategy.go

```diff
@@ -0,0 +1,104 @@
+package balancer
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	NoNodesProvided = "empty slices of nodes"
```

Comment:
uhm... i see no point on these strings being exportable.

I believe it would actually make more sense if these were exportable error types.

Thoughts?